doc/library-reference/inet/mqtt_client.rst

```diff
@@ -10,12 +10,66 @@ MQTT is a publish-subscribe-based lightweight messaging protocol.
           transport layer (normally TCP). That has to be set up using
           channels.
 
+The driver works by running the processing code in a thread which
+communicate with the MQTT broker on one side using channels, and the
+application on the other side using queues.
+
+This means the application has to set up appropriate channels, which
+is already ready to communicate with the MQTT server, e.g. using TCP,
+and the thread running the MQTT client.
```

Comment:
I haven't used to code enough in a real application to be able to tell if the thread is needed enough.

One thing which is in favour of the thread is to be able to keep the MQTT connection alive by sending ping as needed etc. but of course that could be done without e.g. in a polling model.

I have been thinking it might make sense to have two mqtt client - one is the low level (essentially the current one possibly minus thread) and the other one is the high level one that includes "native" TCP/TLS support so it can has less boiler plate for channel setup etc, and support automatic reconnection of the underlying transport... Unfortunately there is finite time so I haven't gotten any further on really thinking about that.